### DISCUSSION TITLE
[@types/d3-shape] stack parameter type seems to be too narrow

### DISCUSSION THREAD

**Discussion by alexejrotar** [DefinitelyTyped/DefinitelyTyped#66042] [Issues with a @types package]
I was trying to create a stacked series, which according to the [d3 docs](https://d3js.org/d3-shape/stack#stack) should work like this:

```javascript
const data = [
  {date: new Date("2015-01-01"), fruit: "apples", sales: 3840},
  {date: new Date("2015-01-01"), fruit: "bananas", sales: 1920},
  {date: new Date("2015-01-01"), fruit: "cherries", sales: 960},
  {date: new Date("2015-01-01"), fruit: "durians", sales: 400},
  {date: new Date("2015-02-01"), fruit: "apples", sales: 1600},
  {date: new Date("2015-02-01"), fruit: "bananas", sales: 1440},
  {date: new Date("2015-02-01"), fruit: "cherries", sales: 960},
  {date: new Date("2015-02-01"), fruit: "durians", sales: 400},
  {date: new Date("2015-03-01"), fruit: "apples", sales: 640},
  {date: new Date("2015-03-01"), fruit: "bananas", sales: 960},
  {date: new Date("2015-03-01"), fruit: "cherries", sales: 640},
  {date: new Date("2015-03-01"), fruit: "durians", sales: 400},
  {date: new Date("2015-04-01"), fruit: "apples", sales: 320},
  {date: new Date("2015-04-01"), fruit: "bananas", sales: 480},
  {date: new Date("2015-04-01"), fruit: "cherries", sales: 640},
  {date: new Date("2015-04-01"), fruit: "durians", sales: 400}
];

const series = d3.stack()
    .keys(d3.union(data.map(d => d.fruit))) // apples, bananas, cherries, …
    .value(([, group], key) => group.get(key).sales)
  (d3.index(data, d => d.date, d => d.fruit));
```

As far as I understand, `d3.index` would return a nested map. if I try this code in TS with @types/d3, the type checker complains, even when explicitly specifying the type parameter `d3.stack<typeof data[number]>`. the resulting function expects an `Iterable<typeof data[number]>`, which I guess is one way of passing in the data. so we could just pass `data` directly rather than `d3.index(...)`, but that produces quite a different series because it is not properly indexed.

In short: I think, it should be possible, to pass indexed data to the stack generator because otherwise, most of its functionality is inaccessible. please correct me, if I got it wrong.

(also, this is my first time posting here, so feel free to make suggestions, how I could improve my question)

[a comment by an automated account is not shown]

**Reply by eft** [reply]
Same experience here.  In the interim, because the functionality of the code appears unaffected, I am suppressing the warnings by preprending

`//@ts-expect-error`

before the line triggering the TS warning
